internal/monitors/manager.go

```diff
@@ -314,26 +314,49 @@ func (mm *MonitorManager) findConfigForMonitorAndRun(endpoint services.Endpoint)
 // endpoint may be nil for static monitors
 func (mm *MonitorManager) createAndConfigureNewMonitor(config config.MonitorCustomConfig, endpoint services.Endpoint) error {
 	id := types.MonitorID(mm.idGenerator())
+	coreConfig := config.MonitorConfigCore()
+	monitorType := coreConfig.Type
 
 	log.WithFields(log.Fields{
-		"monitorType":   config.MonitorConfigCore().Type,
-		"discoveryRule": config.MonitorConfigCore().DiscoveryRule,
+		"monitorType":   monitorType,
+		"discoveryRule": coreConfig.DiscoveryRule,
 		"monitorID":     id,
 	}).Info("Creating new monitor")
 
 	instance := newMonitor(config.MonitorConfigCore().Type, id)
 	if instance == nil {
-		return errors.Errorf("Could not create new monitor of type %s", config.MonitorConfigCore().Type)
+		return errors.Errorf("Could not create new monitor of type %s", monitorType)
+	}
+
+	metadata, ok := MonitorMetadatas[monitorType]
+	if !ok {
+		panic(fmt.Sprintf("could not find monitor metadata of type %s", monitorType))
 	}
 
 	configHash := config.MonitorConfigCore().Hash()
 
+	var extraMetrics []string
+
+	// Monitors can add additional extra metrics to allow through such as based on config flags.
+	if monitorExtra := config.GetExtraMetrics(); monitorExtra != nil {
+		extraMetrics = append(extraMetrics, coreConfig.ExtraMetrics...)
+		extraMetrics = append(extraMetrics, monitorExtra...)
+	} else {
```

Comment:
Any reason why it couldn't do both.  This would just silently ignore `ExtraMetrics` on the config struct this way?

Reply:
Or maybe just have the default implementation of GetExtraMetrics return ExtraMetrics from the config struct.

Reply:
I may be missing your point but I think it's doing both. If `config.GetExtraMetrics()` returns non-nil it merges that and the user-specified `coreConfig.ExtraMetrics`. If the default returns `ExtraMetrics` it puts the onus on the monitor that overrides it (like ES) to do the merging.

Reply:
Ok, in that case this can be simplified to do the handling of config.ExtraMetrics once, without an else block.